src/esm_runscripts/filelists.py

```diff
@@ -938,6 +939,72 @@ def copy_files(config, filetypes, source, target):
     return config
 
 
+def filter_allowed_missing_files(config):
+    """
+    Filters the general.files_missing_when_preparing_run dictionary to move any
+    allowed missing files to a seperate dictionary.
+
+
+    This function can be included to mark specific files as "allowed to be
+    missing". A list of allowed missing files should be put into the model
+    configuration::
+
+        echam:
+            allowed_missing_files:
+                - "*restart*jsbid*"
+                - "unit.*"
+
+    The allowed missing files may either be globs or regular expressions. The
+    filename (without full path) on either the source or the target is used for
+    matching. Therefore, if a file is renamed in the process of being moved
+    from the pool filesystem to the experiment tree, or to the work folder, an
+    exclusion can be made both with the source name or with the target name.
+
+    Parameters
+    ----------
+    config : dict
+        The experiment configuration
+
+    Returns
+    -------
+    config : dict
+    """
+    allowed_missing_files = config["general"].setdefault("allowed_missing_files", {})
+    missing_files = config["general"].get("files_missing_when_preparing_run", {})
+    # TODO(PG): Replace with logger statements
+    print("Currently known missing files:")
+    for k, v in missing_files.items():
+        print(f"source: {k} --> target: {v}")
```

Comment:
Should this print statement by under a `if config["general"].get("verbose", False):`?